src/System.Runtime/tests/System/String.cs

```diff
@@ -337,12 +337,60 @@ public static void TestCompareOrdinal()
 
         i = String.CompareOrdinal("Hello", "Goodbye");
         Assert.True(i > 0);
+    }
 
-        i = String.CompareOrdinal(new String(c), 2, new String(c), 2, 3);
-        Assert.Equal(0, i);
+    [Theory]
+    [InlineData(null, -1, null, -1, -1, 0)]
+    [InlineData("Hello", -1, null, -1, -1, 1)]
+    [InlineData(null, -1, "Hello", -1, -1, -1)]
+    [InlineData("Hello", 0, "Hello", 0, 0, 0)]
+    [InlineData("Hello", 0, "Hello", 0, 5, 0)]
+    [InlineData("Hello", 0, "Hello", 0, 3, 0)]
+    [InlineData("Hello", 2, "Hello", 2, 3, 0)]
+    [InlineData("Hello", 0, "Hello123", 0, int.MaxValue, -1)]           // Recalculated length
+    [InlineData("Hello123", 0, "Hello", 0, int.MaxValue, 1)]            // Recalculated length
+    [InlineData("aaaaaaaaaaaaaa", 3, "aaaaaaaaaaaaaa", 3, 100, 0)]      // Equal long alignment 2, equal compare
+    [InlineData("aaaaaaaaaaaaaa", 3, "aaaxaaaaaaaaaa", 3, 100, -1)]     // Equal long alignment 2, different compare at n=1
+    [InlineData("aaaaaaaaaaaaaa", 1, "aaaaaaaaaaaaaa", 1, 100, 0)]      // Equal long alignment 6, equal compare
+    [InlineData("aaaaaaaaaaaaaa", 1, "axaaaaaaaaaaaa", 1, 100, -1)]     // Equal long alignment 6, different compare at n=1
+    [InlineData("aaaaaaaaaaaaaa", 0, "aaaaaaaaaaaaaa", 0, 100, 0)]      // Equal long alignment 4, equal compare
+    [InlineData("aaaaaaaaaaaaaa", 0, "xaaaaaaaaaaaaa", 0, 100, -1)]     // Equal long alignment 4, different compare at n=1
+    [InlineData("aaaaaaaaaaaaaa", 0, "axaaaaaaaaaaaa", 0, 100, -1)]     // Equal long alignment 4, different compare at n=2
+    [InlineData("aaaaaaaaaaaaaa", 2, "aaaaaaaaaaaaaa", 2, 100, 0)]      // Equal long alignment 0, equal compare
+    [InlineData("aaaaaaaaaaaaaa", 2, "aaxaaaaaaaaaaa", 2, 100, -1)]     // Equal long alignment 0, different compare at n=1
+    [InlineData("aaaaaaaaaaaaaa", 2, "aaaxaaaaaaaaaa", 2, 100, -1)]     // Equal long alignment 0, different compare at n=2
+    [InlineData("aaaaaaaaaaaaaa", 2, "aaaaxaaaaaaaaa", 2, 100, -1)]     // Equal long alignment 0, different compare at n=3
+    [InlineData("aaaaaaaaaaaaaa", 2, "aaaaaxaaaaaaaa", 2, 100, -1)]     // Equal long alignment 0, different compare at n=4
+    [InlineData("aaaaaaaaaaaaaa", 2, "aaaaaaxaaaaaaa", 2, 100, -1)]     // Equal long alignment 0, different compare at n=5
+    [InlineData("aaaaaaaaaaaaaa", 0, "aaaaaaaaaaaaax", 1, 100, -1)]     // Different int alignment
+    [InlineData("aaaaaaaaaaaaaa", 1, "aaaxaaaaaaaaaa", 3, 100, -1)]     // Different long alignment, abs of 4, one of them is 2, different at n=1
+    [InlineData("aaaaaaaaaaaaaa", 1, "aaaaaaaaaaaaax", 4, 100, -1)]     // Different long alignment
```

Comment:
Seems like there are additional cases that would be important...

Maybe I'm missing it, but are there any cases where the entire strings aren't equal but where the substrings specified by the index and offset are?

What about boundary cases, such as where the first characters are different?

Non-ASCII text?

Etc.